contributing/TESTING.md

```diff
@@ -253,55 +251,60 @@ When a test run fails due to visual diffs being present, click the `details` lin
-```
-
-Note that if you drop the `--nobuild` flag, `gulp visual-diff` will run `gulp dist|build --fortesting` on each execution.
+- Navgates to the defined page using a headless Chrome browser
+- Waits for the page to finish loading, both by verifying idle network connections and lack of loader animations
+- If defined, waits until the appropriate CSS selectors appear/disappear from the page
+- If defined, waits an arbitrary amount of time (e.g., for components that have time-delayed mutations)
+- If defined, executes any custom interaction test code
+- Uploads a snapshot of the page's DOM (converted to an HTML string) to the Percy service
 
-The build will use the Percy credentials set via environment variables in the previous step, and run the tests on your local install of Chrome in headless mode. You can see the results at `https://percy.io/<org>/<project>`.
+When all snapshots finish uploading, Percy will process the visual diffs and report back to GitHub as a pull request status. Percy renders the snapshots in their own browsers and take a screenshot. If the new screenshot differs from the previously approved screenshot you will get a visual highlighting of where that difference lies.
 
-To see debugging info during Percy runs, you can run:
-
-```sh
- gulp visual-diff --chrome_debug --webserver_debug
-```
+Percy DOES NOT by default run JavaScript, so the same DOM snapshot will be used in displaying mobile and desktop versions of the page. This means that if your page looks different between these two device types, it should be able to do that using CSS only.
 
-To run tests without uploading snapshots to Percy, you can run:
+### Adding and Modifying Visual Diff Tests
 
-```sh
-gulp visual-diff --percy_disabled
-```
+#### One-time Setup
 
-The debug flags `--chrome_debug` and `--webserver_debug` can be used independently. To enable both debug flags, you can also run:
+First create a free Percy account at [https://percy.io](https://percy.io), create a project, and set the `PERCY_TOKEN` environment variable using the unique value you find at `https://percy.io/<org>/<project>/integrations`:
 
 ```sh
- gulp visual-diff --debug
+export PERCY_TOKEN="<unique-percy-token>"
 ```
 
-To execute only a subset of the tests (i.e., when creating or debugging an existing test) use the `--grep` regular expression flag. e.g., `gulp visual-diff --grep="amp-[a-f]"` will execute on tests that have an AMP component name between `<amp-a...>` through `<amp-f...>`.
-
-After each run, a new set of results will be available at `https://percy.io/<org>/<project>`.
+Once the environment variable is set up, you can run the AMP visual diff tests. You can also pass this token directly to `gulp visual-diff --percy_token="<unique-percy-token>"`
+
+#### Writing the Test
+
+To start, create the page and register it in the configuration file for visual diff tests:
+
+- Create an AMPHTML page that will be tested under `examples/visual-tests`.
```

Comment:
To make this more generally applicable, would it make sense to say "AMP document" instead of "AMPHTML page"?

Reply:
Done